src/StackExchange.Redis/ResultProcessor.cs

```diff
@@ -221,7 +221,7 @@ public virtual bool SetResult(PhysicalConnection connection, Message message, in
             {
                 try
                 {
-                    logging.Log?.WriteLine($"Response from {bridge?.Name} / {message.CommandAndKey}: {result}");
+                    logging.Log?.LogInfo($"Response from {bridge?.Name} / {message.CommandAndKey}: {result}");
```

Comment:
how do we feel about allocations and log-levels? there are the stateful callback APIs that avoid allocs / formats when the log level turns out to be too low, although these take a very different shape, i.e.
```
log.Log(LogLevel.Information, default, (name, id), null, static (state, _) => $"Thing {state.name} of {state.id}");
```

(can be simplified quite a bit by using local extension methods)

if we don't care, that's also fine, as long as deliberate

Reply:
(not sure if possible due to supporting older build SDKs but: https://learn.microsoft.com/en-us/dotnet/core/extensions/logger-message-generator)

Reply:
There are a few ways to make this faster on allocations - totally cool with any of them my thinking is:
1. It should be analyzer enforced if possible
2. It should work everywhere possible.

There exists today the signatures like [LogInformation(ILogger, String, Object[])](https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.logging.loggerextensions.loginformation?view=dotnet-plat-ext-6.0#microsoft-extensions-logging-loggerextensions-loginformation(microsoft-extensions-logging-ilogger-system-string-system-object())) and friends, but if we're formatting that message at all it's the same performance _plus_ the object array in the end to get structured logging.

I was really hoping before tackling this that the interpolated string handler goodness in .NET 6+ (https://devblogs.microsoft.com/dotnet/string-interpolation-in-c-10-and-net-6/) would have landed in logging extension methods but alas this set of APIs didn't happen. I chatted with @stephentoub on this quite a while back and did poke at it, but the fundamental problem you arrive at is "that's not the shape of ILogger" in the end, so if that's your interface someone is calling the formatter in all the cases I could try to be more efficient in. It's entirely possible a smarter API design that I was capable of would still be a win here though, and if so: I'm all game for it.

In the example above my worry cases would be "touching ValueTuple again with a 1000 ft pole" and exceedingly high format count cases to maintain like exception messages today with something like 30-50+ args in play.

I'll keep hacking at making all the basics correct here, but if we can come up with optimal paths to go back through: yeah, absolutely. Do you wanna try a spike against this branch and see what we can do for options? Absolutely wanting to make this more efficient if we reasonably can.